tests/run-make/obey-crate-type-flag/rmake.rs

```diff
@@ -0,0 +1,16 @@
+// test.rs should produce both an rlib and a dylib
+// by default. When the crate_type flag is passed and
+// forced to dylib, no rlibs should be produced.
+// See https://github.com/rust-lang/rust/issues/11573
+
+//@ ignore-cross-compile
+
+use run_make_support::{count_rlibs, remove_dylibs, remove_rlibs, rustc};
+
+fn main() {
+    rustc().input("test.rs").run();
+    remove_rlibs("test");
+    remove_dylibs("test");
```

Comment:
Question: do these pass if nothing are removed? If so, I think that's not very clear and I would prefer to either not have remove operations or to know exactly what is being removed here.